### PR TITLE
mission block: fix incorrectly calculated ccw loiter exit

### PR DESCRIPTION
**Pull request by tstastny**
**Describe problem solved by this pull request**
Addresses #18595
Loiter exit for fixed-wing landing was incorrectly calculated on counter clockwise loiters causing the aircraft to fly an offset approach vector.

**Describe your solution**
Loiter radius was signed in mission item, which means we need to take it's absolute value before doing trig operations (as they were currently implemented).

**Test data / coverage**
SITL with CCW landing. Aborted and let it go around one more time just for good measure.
screen_land_ccw
And in CW direction just to make sure nothing else changed.
scrren_land_cw

@ryanjAA @thomasteisberg

### CODE REVIEW

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -467,8 +467,8 @@ MissionBlock::is_mission_item_reached()
 
 				float bearing = get_bearing_to_next_waypoint(curr_sp.lat, curr_sp.lon, next_sp.lat, next_sp.lon);
 				// We should not use asinf outside of [-1..1].
-				const float ratio = math::constrain(_mission_item.loiter_radius / range, -1.0f, 1.0f);
-				float inner_angle = M_PI_2_F - asinf(ratio);
+				const float ratio = math::min(fabsf(_mission_item.loiter_radius / range), 1.0f);
+				float inner_angle = acosf(ratio);
```
Comment (dagar) [inline src/modules/navigator/mission_block.cpp:471]: Comment above still says asinf.

**Comment by ryanjAA** [comment]
Great to know.  Thanks! 

Checked back on previous release 1.11.3 and this works fine so will see when occurred. 

Awesome!

File: src/modules/navigator/mission_block.cpp (hunk 1 above)
Comment (tstastny) [reply src/modules/navigator/mission_block.cpp:472]: thanks for the catch

**Review by sfuhrer — DISMISSED** [review]
Good catch!

**Review by sfuhrer — APPROVED** [review]

**Pull request merged**